### PR TITLE
Add support for gitlab branch protection

### PR DESCRIPTION
**Pull request by waseem-h**
I've added gitlab branch protection support since it was needed for our usecase and others might also require it. We're using a local binary of this instead of the global release for now. We want it to be merged so that our dependency on a local binary is removed.

### CODE REVIEW

**Comment by waseem-h** [comment]
@radeksimko @grubernaut @roidelapluie Please take a look

**Comment by waseem-h** [comment]
@radeksimko @grubernaut @roidelapluie Please review

**Comment by grubernaut** [comment]
hey @waseem-h would love to, but I left Hashicorp over a year ago. Will likely need a current employee to review this for you. :smile:

**Comment by roidelapluie** [comment]
Please add acceptance tests :)

**Comment by roidelapluie** [comment]
Any news?

**Comment by roidelapluie** [comment]
Missing:

- Documentation
- Acceptance test

Thanks!!

File: gitlab/resource_gitlab_branch_protection.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabBranchProtection() *schema.Resource {
+	acceptedAccessLevels := []string{
```
Comment (mcanevet) [inline gitlab/resource_gitlab_branch_protection.go]: How could I set access level to "No one"?

**Comment by mcanevet** [comment]
Once https://github.com/terraform-providers/terraform-provider-gitlab/pull/83 is merged, we should be able to add `no one` to `acceptedAccessLevels`.

**Comment by roidelapluie** [comment]
#83 is tested but there is still no acceptance tests

File: gitlab/resource_gitlab_branch_protection.go (hunk 1 above)
Comment (roidelapluie) [reply gitlab/resource_gitlab_branch_protection.go]: is it possible to get that list from elsewhere?

Comment (mcanevet) [reply gitlab/resource_gitlab_branch_protection.go]: Could you please add `no one` and `maintainer` here?

Comment (ahmadiq) [reply gitlab/resource_gitlab_branch_protection.go]: > is it possible to get that list from elsewhere?

@roidelapluie should we sync the branch with the upstream master and then use the accessLevel map from https://github.com/terraform-providers/terraform-provider-gitlab/blob/master/gitlab/util.go#L140

**Comment by bilby91** [comment]
@waseem-h Any plans on adding tests for this one ? I can create another PR for this one if not.

**Comment by ahmadiq** [comment]
> @waseem-h Any plans on adding tests for this one ? I can create another PR for this one if not.

@bilby91 we are planning this for the coming week

**Comment by usamaahmadkhan** [comment]
@roidelapluie there is a 401 error in test case 

> TestAccDataGitlabProject_basic

which is causing the pipeline to fail subsequently. Can you please look into this?

**Comment by usamaahmadkhan** [comment]
@roidelapluie @bilby91  @mcanevet can you plz take a look?

**Comment by roidelapluie** [comment]
pending #118 is merged for green CI test

**Pull request merged**